### PR TITLE
Feature/5297 5368 event checkin persistence mock

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Adds the API for Event and Checkin persistence.
Adds mock implementation of the API.

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+enum EventType {
+	case type1
+	case type2
+	case type3
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift:8]: Can you add some comments or make the cases more precise?

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+struct Event {
+
+	// The ID of the event. Note that the ID is generated by the CWA server. It is stored as base64-encoded string of the guid attribute of Protocol Buffer message Event.
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift:13]: SwiftDoc
```suggestion
	/// The ID of the event. Note that the ID is generated by the CWA server. It is stored as base64-encoded string of the guid attribute of Protocol Buffer message Event.
```

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	let id: String
+	let type: EventType
+	let description: String
+	let address: String
+	let start: Date
+	let end: Date
+	let defaultCheckInLengthInMinutes: Int
+	// The signature of the event (provided by the CWA server). It is stored as a base64-encoded string of the signature attribute of Protocol Buffer message SignedEvent.
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift:21]:
```suggestion
	/// The signature of the event (provided by the CWA server). It is stored as a base64-encoded string of the signature attribute of Protocol Buffer message SignedEvent.

```

File: src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift (hunk 1 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Event.swift:8]: At the moment not, maybe in the near future.

**Comment by ArturFriesen** [comment]
Spec changed. New implementation needed.

**Pull request closed without merging**